### PR TITLE
validation for agenda

### PR DESCRIPTION
**Pull request by Maroju100**
Added validations to agenda

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemController.java [hunk 1]
```diff
@@ -40,105 +42,58 @@
...
     /**
-     * Load agenda items
+     * Get agenda item from id
      *
-     * @param agendaItems, {@link List< AgendaItemCreateDTO > to load {@link AgendaItem agenda items}}
-     * @return {@link HttpResponse<List< AgendaItem >}
+     * @param id
+     * @return
      */
-    @Post("/items")
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemController.java:118]: why was this end point deleted. ?

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java [hunk 2]
```diff
@@ -1,71 +1,115 @@
...
 
-import static com.objectcomputing.checkins.services.role.RoleType.Constants.MEMBER_ROLE;
+import static com.objectcomputing.checkins.services.role.RoleType.Constants.*;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 
-class AgendaItemControllerTest extends TestContainersSuite implements MemberProfileFixture, CheckInFixture, AgendaItemFixture {
+public class AgendaItemControllerTest extends TestContainersSuite implements MemberProfileFixture, CheckInFixture, AgendaItemFixture {
```
Comment (bhatiam12) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:29]: There are no tests for delete end point. please add those.

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImplTest.java [hunk 3]
```diff
@@ -1,331 +1,331 @@
-package com.objectcomputing.checkins.services.agenda_item;
+// package com.objectcomputing.checkins.services.agenda_item;
 
-import com.objectcomputing.checkins.services.checkins.CheckIn;
```
Comment (bhatiam12) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImplTest.java:3]: delete this class if none of the tests are being used and make sure you have covered everything in service implementation via end to end tests.

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java [hunk 4]
```diff
@@ -1,82 +1,158 @@
...
+
+        return agendaItem;
     }
 
+    @Override
     public void delete(@NotNull UUID id) {
-        agendaItemRepo.deleteById(id);
-    }
-}
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
```
Comment (meyerinj) [inline server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java:136]: There are no tests exercising this method

**Review by meyerinj — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemController.java (hunk 1 above)
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemController.java]: thanks, added back, got removed somehow.

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java (hunk 2 above)
Comment (Maroju100) [reply server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:29]: Added test for delete endpoint. Thanks

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImplTest.java (hunk 3 above)
Comment (Maroju100) [reply server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImplTest.java:3]: Deleted the file, covered in end to end tests.

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemBadArgException.java [hunk 5]
```diff
@@ -1,7 +1,12 @@
 package com.objectcomputing.checkins.services.agenda_item;
 
 public class AgendaItemBadArgException extends RuntimeException {
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
```
[the remark this answers is not in the document]
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemBadArgException.java:5]: Updated, thanks.

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java (hunk 4 above)
Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java:136]: Added the tests for delete.

File: server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java [hunk 6]
```diff
@@ -1,82 +1,160 @@
...
-            } else if (!memberRepo.findById(createById).isPresent()) {
-                throw new AgendaItemBadArgException(String.format("Member %s doesn't exist", createById));
+            CheckIn checkinRecord = checkinRepo.findById(checkinId).orElse(null);
+            Boolean isCompleted = checkinRecord != null ? checkinRecord.isCompleted() : null;
+            final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+            validate(checkinId == null || createById == null, "Invalid agenda item %s", agendaItem);
+            validate(agendaItem.getId() != null, "Found unexpected id %s for agenda item", agendaItem.getId());
+            validate(checkinRepo.findById(checkinId).isEmpty(), "CheckIn %s doesn't exist", checkinId);
+            validate(memberRepo.findById(createById).isEmpty(), "Member %s doesn't exist", createById);
+            if (!isAdmin && isCompleted) {
```
Comment (Bagurp) [inline server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java:56]: If user is not admin and if checkin is complete, save operation is not allowed. Shouldn't we throw exception directly for this?

Comment (Maroju100) [reply server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java:56]: updated to throw exception directly.

**Review by Bagurp — APPROVED** [review]

**Pull request merged**